### PR TITLE
feature: add configuration file for functional tests

### PR DESCRIPTION
**Pull request by h4sh3d**
Configuration can now be done with a config file. Other file renaming happen in this PR.

### CODE REVIEW

File: tests/functional.rs [hunk 1]
```diff
@@ -1698,13 +1701,17 @@ async fn monero_syncer_broadcast_tx_test() {
...
-    let whost = env::var("MONERO_WALLET_HOST_1").unwrap_or("localhost".into());
-    let wallet_client = monero_rpc::RpcClient::new(format!("http://{}:18083", whost));
-    let wallet = wallet_client.wallet();
+    let conf = config::TestConfig::parse();
+
+    let client = monero_rpc::RpcClient::new(format!("http://{}", conf.monero.daemon));
+    let regtest = client.daemon().regtest();
+
+    let client = monero_rpc::RpcClient::new(format!(
+        "http://{}",
```
Comment (TheCharlatan) [inline tests/functional.rs:1710]: I think the url should be moved to the config as well. Might save a line here and there and to me it feels like it is part of the configuration.

File: tests/cfg/electrs.toml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# DO NOT EDIT THIS FILE DIRECTLY - COPY IT FIRST!
+# If you edit this, you will cry a lot during update and will not want to live anymore!
```
Comment (TheCharlatan) [inline tests/cfg/electrs.toml:2]: lol

**Review by TheCharlatan — CHANGES_REQUESTED** [review]

File: tests/functional.rs (hunk 1 above)
Comment (h4sh3d) [reply tests/functional.rs]: Done in b16d35c9d53330344a0a101502ef4bea0788a3e6

**Review by TheCharlatan — APPROVED** [review]
lgtm :)

**Pull request merged**